src/libs/OptionsListUtils.js

```diff
@@ -667,7 +667,8 @@ function sortCategories(categories) {
      * }
      */
     _.each(categories, (category) => {
-        const path = category.name.replaceAll(CONST.PARENT_CHILD_SEPARATOR, '.');
+        const safeCategoryName = category.name.replaceAll(CONST.PATH_DOT, CONST.PATH_DOT_REPLACER);
```

Comment:
While it's great that we handle the "Mr. Nobody" case now, I'm not convinced by the solution itself. Why do we insist on using the string / dot-based variant of the `path` argument of `lodashSet`? It's not suitable for any case where parts of the path are externally provided strings.

The [docs](https://lodash.com/docs/4.17.15#set) state that `lodashSet` can accept an array. Splitting on `PARENT_CHILD_SEPARATOR` feels like the cleanest solution possible.

Just let's hope nobody will want to have a category tree like "Movies" > "Avengers: Endgame" 😳

Reply:
I'll start testing this PR in the app, and will check if "Movies" > "Avengers: Endgame" is a possible case

Reply:
The array like a path sounds good. I will try to implement it. Thank you!

Reply:
Done - 0f8cfb5.